lib/rspec/core/configuration.rb

```diff
@@ -828,6 +828,18 @@ def profile_examples
       end
 
       # @private
+      def profiler
+        @profiler ||=
+          begin
+            if value_for(:profile_examples) { @profile_examples }
+              Profiler.new
+            else
+              NoProfiler.new
+            end
+          end
+      end
```

Comment:
There's a bug here...consider what would happen here:

``` ruby
config.profile_examples = true
config.profiler
config.profile_examples = false
config.profiler
```

Or:

``` ruby
config.profile_examples = false
config.profiler
config.profile_examples = true
config.profiler
```

In either case, the value of `config.profiler` does not update to reflect the mutated value of `profile_examples`.  Our normal way to deal with this is to set `@profiler = nil` in `profile_examples=` so that it clears it out if it changes.